### ISSUE TITLE
Ability to add Place in the StoryMap chapter interface

### ISSUE THREAD

**Issue by martenwilmink** [martenwilmink/reframe-brain#12]
## Issue description [(i)](## "A (short) description as to what the issue is.")

Currently Places are managed separately in the MODx Manager through its own admin menu / page. It would be very user friendly to allow Places to be added right in the StoryMap chapter ContentBlock on the page.

## New specification [(i)](## "A clear and unambiguous extended specification of the new or updated functionality. References to external documents should be temporary and replaced with contents before the ticket is taken up by a developer.")

- StoryMap Chapter shows button to Add Place besides dropdown selector for selecting Existing Place;
- Clicking Add Place opens the 'Edit Place' modal window to create a new Place;
- After entering info and clicking the 'Save and Close' button, the form input gets saved as a new Place to the database and gets selected in the Existing Places dropdown;
- Places can still be managed via the Places menu item in the main menu;
- Stories option and tab in main menu can be removed, so it just shows Places;

## New design [(i)](## "A clear and unambiguous extended design of the new functionality. References to external documents should be temporary and replaced with contents before the ticket is taken up by a developer.")

<img width="1050" alt="rb-12-addPlace" src="https://github.com/user-attachments/assets/0a8386b2-0ba7-47b3-bf28-e99c69b0abf0">

- We should update the description to read something like "Select an exisiting place to connect to this chapter or click Add Place to create a new location"

<img width="778" alt="rb-12-addPlace-modal" src="https://github.com/user-attachments/assets/c20b25d9-c381-4d41-a898-233c1d3ed387">

## Existing specification [(i)](## "Link(s) to the specification regarding the relevant existing functionality. The links or surrounding text should clearly indicate what is being referenced, so the information can also be found if the link goes 404.")
- N/A (@hugopeek you can link/reference it here if it does exist)

## Existing design [(i)](## "If the issue concerns visual matters at all, link(s) to the designs regarding the relevant existing functionality. The links or surrounding text should clearly indicate what is being referenced, so the information can also be found if the link goes 404.")

<img width="953" alt="rb-12-existing-Places" src="https://github.com/user-attachments/assets/d858c79d-8d35-46f7-a517-3b4e62bdb582">

## Additional information [(i)](## "Suggestions for technical implementation, solution pointers, information that provides extra context or background information, related issues, comments, etc.")
- Goldplating: replace the MIGX title when editing/viewing Places with... Places
- We could consider adding a link / button to the Places overview page, e.g. 'Manage Places' in the description

## TODO  [(i)](## "Checklist to ensure critical reviews of specification are done by issue creator and lead developer before actual development starts.")
- [x] Self review of issue specification
- [x] Lead dev review of issue specification

**Comment by martenwilmink** [comment]
@hugopeek did you have this issue on your radar still too? Would be nice to get this functionality added before I showcase another demo.

**Comment by hugopeek** [comment]
@martenwilmink This is in the works now!

**Comment by hugopeek** [comment]
@martenwilmink This turned out to be trickier than expected. The idea was to clone the Add Place button from the MIGX TV, but there are a few problems with that:

- The MIGX TV is only initialized *after* the page is loaded, so I couldn't simply pluck the button out of the DOM with a plugin.
- Same problem with ContentBlocks: initialized dynamically after DOM is loaded, but at least it attaches a class to the body once it's done.
- Once properly targeted, it proved to be impossible to clone the MIGX button while keeping the ExtJS events intact.
- I then tried to simply simulate a click on the Add Place button (as one might do with automated frontend tests). That took some fiddling, but I got that to work..
- I then thought: why not hijack the edit button too, so you can also edit a selected place from the CB interface, but that quickly descended into a terrifying ExtJS abyss. It does things like calculating mouse position with x/y coordinates.. Absolute madness, literally! One does not simply simulate a click action on an ExtJS element inside a grid...
- Getting the value of the newly created Place back and applying that to the CB settings selector is another potential rabbit hole. CB is *not* using ExtJS, so some magic with session storage or something similar would be required.
- That same CB selector is also not getting refreshed after a place is added. Only the MIGX grid is. I thought of triggering a simple browser tab refresh with a return to the same scroll position, but ExtJS also messes with that..

Long story short: the Add Place button is being inserted below the CB selector, but all it can do is just that: add a place. It won't show up in the selector until after a refresh. I'm still thinking about the quickest way to fix at least this behaviour. Then I would consider it to be a useful ~~feature~~ hack.

As an aside: while diving into the ContentBlocks dev resources, I stumbled upon an announcement of CB 2.0. It seems Mark also has plans to make interacting with the blocks inside the manager a lot easier!

And aside 2: the Stories grid was removed from the Reframe CMP. It is now just Places.
